### PR TITLE
Cls2 232 link subsidiary button

### PR DESCRIPTION
**Pull request by CaitBarnard**
## Description of change

Button to link to 'add new subsidiary' page. This will only appear if a company already has manually linked subsidiaries in the tree.

## Test instructions

_What should I see?_

### After
Screenshot 2023-06-29 at 11 23 21

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [ ] Has the branch been rebased to main?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

**Review by dredmonds — APPROVED** [review]

File: src/client/modules/Companies/CompanyHierarchy/styled.jsx [hunk 1]
```diff
@@ -63,6 +63,35 @@ export const SubsidiaryList = styled.ul`
 `
 export const StyledButton = styled(Button)``
 
+export const StyledLinkedSubsidiaryButton = styled(Button)`
+  margin-top: 20px;
+  transform-style: preserve-3d;
+
+  // This is the horizontal line to the left of the button
```
Comment (chopkinsmade) [inline src/client/modules/Companies/CompanyHierarchy/styled.jsx:70]: As this is copied from the `HierarchyItemContents` component, can this be moved into a variable in this file and reused in both?

File: src/client/modules/Companies/CompanyHierarchy/styled.jsx [hunk 2]
```diff
@@ -63,6 +63,35 @@ export const SubsidiaryList = styled.ul`
...
+    width: 100px;
+    height: 5px;
+    top: 16px;
+    left: -40px;
+    display: block;
+    transform: translateZ(-1px);
+  }
+  // This is the vertical line that stretches between the last company and the button
+  :after {
+    content: '';
```
Comment (chopkinsmade) [inline src/client/modules/Companies/CompanyHierarchy/styled.jsx:84]: Same as above, although it doesn't look like an exact copy as the left is different. We can still have the common bits in a variable though

File: src/client/modules/Companies/CompanyHierarchy/CompanyTree.jsx [hunk 3]
```diff
@@ -87,10 +89,24 @@ const Subsidiaries = ({
...
               requestedCompanyHasManuallyVerified && hierarchy === 1
                 ? false
-                : index + 1 === company.subsidiaries.length
+                : index + 1 === company.subsidiaries.length && !isManuallyLinked
             }
           />
         ))}
+        {isManuallyLinked && (
+          <li>
+            <StyledLinkedSubsidiaryButton
```
Comment (chopkinsmade) [inline src/client/modules/Companies/CompanyHierarchy/CompanyTree.jsx:98]: Using a button with an onClick removes the ability to use react-router. Instead use a button as a link so you can specify the href. If you search for `as={Link}` in the code you will find lots of examples, this page has some src/apps/companies/apps/referrals/details/client/ReferralDetails.jsx

File: src/client/modules/Companies/CompanyHierarchy/CompanyTree.jsx [hunk 4]
```diff
@@ -87,10 +89,24 @@ const Subsidiaries = ({
...
+              buttonColour={GREY_4}
+              buttonTextColour={BLACK}
+              data-test="link-subsidiary-button"
+              onClick={() =>
+                (window.location = `/companies/${requestedCompanyId}/subsidiaries/link`)
```
Comment (chopkinsmade) [inline src/client/modules/Companies/CompanyHierarchy/CompanyTree.jsx:103]: Use the urls module to generate a link, instead of writing it yourself

**Review by chopkinsmade — APPROVED** [review]

**Pull request merged**